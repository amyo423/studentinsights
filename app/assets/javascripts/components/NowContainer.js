import React from 'react';

// A container for testing that allows passing in a `nowFn`
// that will be shared with child components through context.
export default class NowContainer extends React.Component {
  getChildContext() {
    const {nowFn} = this.props;
    return {
      nowFn() { return nowFn(); }
    };
  }

  render() {
    const {children} = this.props;

    return children;
  }
}
NowContainer.childContextTypes = {
  nowFn: React.PropTypes.func
};
NowContainer.propTypes = {
  children: React.PropTypes.node.isRequired,
  nowFn: React.PropTypes.func.isRequired
};