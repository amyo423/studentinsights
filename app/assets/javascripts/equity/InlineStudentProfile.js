import React from 'react';
import _ from 'lodash';
import SectionHeading from '../components/SectionHeading';
import GenericLoader from '../components/GenericLoader';
import FeedView from '../feed/FeedView';


// Inline student profile for classroom list creator, shown as a modal
export default class InlineStudentProfile extends React.Component {
  render() {
    const {student} = this.props;
    return (
      <div className="InlineStudentProfile">
        <h4><a style={{fontSize: 20}} href={`/students/${student.id}`} target="_blank">{student.first_name} {student.last_name}</a></h4>
        <div style={styles.columns}>
          <div style={styles.column}>
            <div>Disability: {student.disability}</div>
            <div>Program: {student.program_assigned}</div>
            <div>504 plan: {student.plan_504}</div>
            <div>Learning English: {student.limited_english_proficiency}</div>
            <div>Most recent ACCESS: <pre>{JSON.stringify(_.last(student.latest_access_results), null, 2)}</pre></div>
            <div>Home language: {student.home_language}</div>
            <br />
            <div>Free reduced lunch: {student.free_reduced_lunch}</div>
            <div>Race: {student.race}</div>
            <div>Hispanic: {student.hispanic_latino ? 'yes' : 'no'}</div>
            <div>Gender: {student.gender}</div>
            <br />
            <div>Most recent DIBELS: {student.latest_dibels && student.latest_dibels.performance_level}</div>
            <div>Most recent STAR Math percentile: {student.most_recent_star_math_percentile}</div>
            <div>Most recent STAR Reading percentile: {student.most_recent_star_reading_percentile}</div>
          </div>
          <div style={styles.column}>
            {this.renderNotes()}
          </div>
        </div>
      </div>
    );
  }

  // TODO(kr) refactor
  renderNotes() {
    const {student, fetchProfile} = this.props;
    return (
      <div>
        <SectionHeading>Notes for {student.first_name}</SectionHeading>
        <GenericLoader
          promiseFn={() => fetchProfile(student.id)}
          render={this.renderFeed.bind(this)} />
      </div>
    );
  }

  renderFeed(json) {
    return <FeedView feedCards={json.feed_cards} />;
  }

  // // TODO(kr) refactor
  // renderNotesFeed(json) {
  //   const {student} = this.props;
  //   const feed = {
  //     event_notes: student.event_notes_without_restricted,
  //     services: {
  //       active: [],
  //       discontinued: []
  //     },
  //     deprecated: {
  //       interventions: []
  //     }
  //   };

  //   // TODO(kr) module
  //   return (
  //     <window.shared.NotesList
  //       feed={feed}
  //       educatorsIndex={json.educators_index}
  //       eventNoteTypesIndex={json.event_note_types_index} />
  //   );
  // }
}
InlineStudentProfile.propTypes = {
  fetchProfile: React.PropTypes.func.isRequired,
  student: React.PropTypes.object.isRequired
};

const styles = {
  columns: {
    marginTop: 20,
    display: 'flex',
    flexDirection: 'column'
  },
  column: {
    flex: 1,
  }
};