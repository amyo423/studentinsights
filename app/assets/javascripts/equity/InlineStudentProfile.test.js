import React from 'react';
import ReactDOM from 'react-dom';
import {withDefaultNowContext} from '../../../../spec/javascripts/support/NowContainer';
import InlineStudentProfile from './InlineStudentProfile';
import profile_json from './fixtures/profile_json';
import students_for_grade_level_next_year_json from './fixtures/students_for_grade_level_next_year_json';


export function testProps(props) {
  return {
    fetchProfile(studentId) {
      return Promise.resolve(profile_json);
    },
    student: students_for_grade_level_next_year_json.students[0],
    ...props
  };
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  const props = testProps();
  ReactDOM.render(withDefaultNowContext(<InlineStudentProfile {...props} />), el);
});
