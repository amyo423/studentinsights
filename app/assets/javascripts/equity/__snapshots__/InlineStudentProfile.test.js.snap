// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  className="InlineStudentProfile"
  style={
    Object {
      "margin": 20,
    }
  }
>
  <h4>
    <a
      href="/students/144"
      style={
        Object {
          "fontSize": 24,
        }
      }
      target="_blank"
    >
      Elsa
       
      Skywalker
    </a>
  </h4>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "marginTop": 20,
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          className="Card "
          style={
            Object {
              "border": "1px solid #ccc",
              "borderRadius": 3,
              "flex": 1,
              "margin": 5,
              "padding": 10,
            }
          }
        >
          <div
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Special needs
          </div>
          <div>
            Disability: 
            Specific LDs
             (
            <a
              href="#"
            >
              IEP
            </a>
            )
          </div>
          <div>
            Program: 
            Sp Ed
          </div>
        </div>
        <div
          className="Card "
          style={
            Object {
              "border": "1px solid #ccc",
              "borderRadius": 3,
              "flex": 1,
              "margin": 5,
              "padding": 10,
            }
          }
        >
          <div
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Learning English
          </div>
          <div>
            Learning English: 
            Fluent
          </div>
          <div>
            ACCESS Composite: 
            1
          </div>
          <div>
            Home language: 
            Portuguese
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          className="Card "
          style={
            Object {
              "border": "1px solid #ccc",
              "borderRadius": 3,
              "flex": 1,
              "margin": 5,
              "padding": 10,
            }
          }
        >
          <div
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Identity
          </div>
          <div>
            Racial identity: 
            White
          </div>
          <div>
            Hispanic: 
            Not hispanic
          </div>
          <div>
            Gender: 
            F
          </div>
        </div>
        <div
          className="Card "
          style={
            Object {
              "border": "1px solid #ccc",
              "borderRadius": 3,
              "flex": 1,
              "margin": 5,
              "padding": 10,
            }
          }
        >
          <div
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Home
          </div>
          <div>
            Home language: 
            Portuguese
          </div>
          <div>
            Free reduced lunch: 
            Not Eligible
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          className="Card "
          style={
            Object {
              "border": "1px solid #ccc",
              "borderRadius": 3,
              "flex": 1,
              "margin": 5,
              "padding": 10,
            }
          }
        >
          <div
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Standardized tests
          </div>
          <div>
            DIBELS: 
            Intensive
          </div>
          <div>
            STAR Math percentile: 
            47
          </div>
          <div>
            STAR Reading percentile: 
            36
          </div>
        </div>
        <div
          className="Card "
          style={
            Object {
              "border": "1px solid #ccc",
              "borderRadius": 3,
              "flex": 1,
              "margin": 5,
              "padding": 10,
            }
          }
        >
          <div
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Behavior
          </div>
          <div>
            Discipline incidents: 
            1
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div>
        <div
          className="SectionHeading"
          style={
            Object {
              "borderBottom": "1px solid #333",
              "padding": 10,
            }
          }
        >
          <h4
            style={
              Object {
                "color": "black",
                "display": "inline",
              }
            }
          >
            Notes for 
            Elsa
          </h4>
        </div>
        <div
          style={
            Object {
              "padding": 10,
            }
          }
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</div>
`;
